src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,52 +40,1012 @@ const ActorInit En_Niw_InitVars = {
+
+        // clang-format off
+        this->niwStormActive = this->unusedCounter28C = this->unk292 = this->unk29E = this->unk298 = this->unk29A = this->unk29C = 0;
+        // clang-format on
+
+        for (i = 0; i < 0xA; ++i) {
+            this->unk264[i] = 0.0f;
+        }
+
+        this->unknownState28E = 8;
+        this->niwStormActive = false;
+        this->actionFunc = EnNiw_LandBeforeIdle;
+        return; // still required even with the else/else
+
+    } else if (((this->actor.bgCheckFlags & 0x20)) && (this->actor.yDistToWater > 15.0f) &&
+               (this->unknownState28E != 6)) {
+        this->actor.velocity.y = 0.0f;
+        this->actor.gravity = 0.0f;
+        Math_Vec3f_Copy(&pos, &this->actor.world.pos);
+        pos.y += this->actor.yDistToWater;
+        this->unkTimer250 = 30;
+        EffectSsGSplash_Spawn(globalCtx, &pos, 0, 0, 0, 400);
+        this->unkTimer252 = 0;
+        this->unknownState28E = 6;
+        this->actionFunc = EnNiw_Swimming;
+
+    } else {
+
+        if ((this->niwStormActive) && (this->actor.xyzDistToPlayerSq < (SQ(dist))) && (player->invincibilityTimer == 0)) {
+            func_800B8D50(globalCtx, &this->actor, 2.0f, this->actor.world.rot.y, 0.0f, 0x10);
+        }
+
+        EnNiw_CheckRage(this, globalCtx);
+        if ((this->flutterSfxTimer == 0) && (this->unknownState28E == 4)) {
+            this->flutterSfxTimer = 7;
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_CHICKEN_FLUTTER);
+        }
+
+        if (this->sfxTimer1 == 0) {
+            if (this->unknownState28E != 0) {
+                this->sfxTimer1 = 30;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_A); // attack cluck
+            } else {
+                this->sfxTimer1 = 300;
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_N); // cluck
+            }
+        }
+
+        if (!this->niwStormActive) {
+            if (this->niwType == ENNIW_TYPE_REGULAR) {
+                Collider_UpdateCylinder(&this->actor, &this->collider);
+                CollisionCheck_SetAC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+
+                if (globalCtx) {} // req to match
+
+                if ((this->unknownState28E != 4) && (this->unknownState28E != 5)) {
+                    CollisionCheck_SetOC(globalCtx, &globalCtx->colCheckCtx, &this->collider.base);
+                }
+            }
+        }
+    }
+}
+
+s32 EnNiw_LimbDraw(GlobalContext* gCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, struct Actor* actor) {
+    EnNiw* this = (EnNiw*)actor;
+
+    // float to s16 casts req
+    if (limbIndex == 0xD) {
+        rot->y += (s16)this->limbDRot;
+    }
+    if (limbIndex == 0xF) {
+        rot->y += (s16)this->limbFRot;
+    }
+    if (limbIndex == 0xB) {
+        rot->x += (s16)this->limbBRotx;
+        rot->y += (s16)this->limbBRoty;
+        rot->z += (s16)this->limbBRotz;
+    }
+    if (limbIndex == 0x7) {
```

Comment:
```suggestion
    if (limbIndex == 13) {
        rot->y += (s16)this->limbDRot;
    }
    if (limbIndex == 15) {
        rot->y += (s16)this->limbFRot;
    }
    if (limbIndex == 11) {
        rot->x += (s16)this->limbBRotx;
        rot->y += (s16)this->limbBRoty;
        rot->z += (s16)this->limbBRotz;
    }
    if (limbIndex == 7) {
```
limbIndex is usually in decimal